File: tourism-system-java/src/main/java/com/tourism/model/entity/SpotPlace.java
package com.tourism.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("spot_place")
public class SpotPlace {

    /** 原始字符串ID，如 "place_001" */
    @TableId(type = IdType.INPUT)
    private String id;

    private String name;

    /** 类型：校园 / 景区 / 公园 等 */
    private String type;

    /** 关键词，JSON数组 */
    private String keywords;

    /** 特色标签，JSON数组 */
    private String features;

    private BigDecimal rating;

    private Integer ratingCount;

    private Integer clickCount;

    /** 纬度 */
    private BigDecimal lat;

    /** 经度 */
    private BigDecimal lng;

    private String address;

    private String openTime;

    private String image;

    @TableField(value = "`description`")
    private String description;

    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: TourismSystemApp/TourismSystemApp/Models/SpotPlace.swift
import Foundation

struct SpotPlace: Identifiable, Codable, Hashable {
    let id: String
    let name: String?
    let type: String?
    let keywords: String?
    let features: String?
    let rating: Double?
    let ratingCount: Int?
    let clickCount: Int?
    let lat: Double?
    let lng: Double?
    let address: String?
    let openTime: String?
    let image: String?
    let description: String?
    let createdAt: String?
    let updatedAt: String?

    var parsedKeywords: [String] { parseJSONArray(keywords) }
    var parsedFeatures: [String] { parseJSONArray(features) }
    var ratingValue: Double { rating ?? 0 }
    var coordinate: (Double, Double)? {
        guard let lat, let lng else { return nil }
        return (lat, lng)
    }

    func hash(into hasher: inout Hasher) { hasher.combine(id) }
    static func == (lhs: SpotPlace, rhs: SpotPlace) -> Bool { lhs.id == rhs.id }
}

func parseJSONArray(_ raw: String?) -> [String] {
    guard let raw, let data = raw.data(using: .utf8),
          let arr = try? JSONDecoder().decode([String].self, from: data) else { return [] }
    return arr
}
